universal-application-tool-0.0.1/app/controllers/dev/FileUploadController.java

```diff
@@ -0,0 +1,82 @@
+package controllers.dev;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import java.io.FileInputStream;
+import java.nio.file.Files;
+import java.util.Comparator;
+import java.util.Set;
+import models.StoredFile;
+import play.Environment;
+import play.libs.Files.TemporaryFile;
+import play.mvc.Controller;
+import play.mvc.Http.MultipartFormData;
+import play.mvc.Http.Request;
+import play.mvc.Result;
+import repository.StoredFileRepository;
+import views.dev.FileUploadView;
+
+/** Controller for interacting with S3 directly in dev mode. */
+public class FileUploadController extends Controller {
+  private final Environment environment;
+  private final FileUploadView view;
+  private final StoredFileRepository storedFileRepository;
+
+  @Inject
+  public FileUploadController(
+      FileUploadView view, StoredFileRepository storedFileRepository, Environment environment) {
+    this.view = checkNotNull(view);
+    this.storedFileRepository = checkNotNull(storedFileRepository);
+    this.environment = checkNotNull(environment);
+  }
+
+  public Result index(Request request) {
+    if (environment.isDev()) {
```

Comment:
suggestion (non-blocking): prefer an early return with `!` in the predicate rather than `if/else`

Reply:
Done.